permits/views.py

```diff
@@ -609,6 +652,57 @@ def handle_prolongation_form_submission(self, form):
         else:
             return redirect("permits:permit_requests_list")
 
+    def handle_complementary_documents_form_submission(self, form):
+        for f in form:
+            f.instance.owner = self.request.user
+            f.instance.permit_request = self.permit_request
+            f.save()
+
+        success_message = (
+            _("Les documents ont bien été ajoutés à la demande #%s.")
+            % self.permit_request.pk
+        )
+        messages.success(self.request, success_message)
+
+        if "save_continue" in self.request.POST:
+            return redirect(
+                "permits:permit_request_detail",
+                permit_request_id=self.permit_request.pk,
+            )
+        else:
+            return redirect("permits:permit_requests_list")
+
+
+class PermitRequestComplementaryDocumentDeleteView(DeleteView):
+    model = models.PermitRequestComplementaryDocument
+    success_message = _("Le document '%s' a été supprimé avec succès")
+    final_error_message = _("Les documents finaux ne peuvent pas être supprimés")
+    owner_error_message = _("Vous pouvez seulement supprimer vos documents")
+
+    def get(self, request, *args, **kwargs):
+        return self.post(request, *args, **kwargs)
+
+    def delete(self, request, *args, **kwargs):
+        obj = self.get_object()
+
+        if not obj.owner == self.request.user:
```

Comment:
I would check with the client with this, but I would let a superuser delete the documents too, with:
`if not obj.owner == self.request.user or self.request.user.is_superuser:`